library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/MaskAllNumberPickerMapperTest.kt

```diff
@@ -0,0 +1,655 @@
+        )
+        val expectedNextLabelWireframe = MobileSegment.Wireframe.TextWireframe(
+            id = fakeNextLabelId,
+            x = fakeExpectedNextLabelXPos,
+            y = fakeExpectedNextLabelYPos,
+            width = fakeExpectedLabelWidth,
+            height = fakeExpectedLabelHeight,
+            textStyle = MobileSegment.TextStyle(
+                family = NumberPickerMapper.FONT_FAMILY,
+                size = fakeExpectedTextSize,
+                color = fakeExpectedNextPrevLabelHtmlColor
+            ),
+            textPosition = MobileSegment.TextPosition(
+                alignment = MobileSegment.Alignment(
+                    horizontal = MobileSegment.Horizontal.CENTER,
+                    vertical = MobileSegment.Vertical.CENTER
+                )
+            ),
+            text = expectedNextLabelValue
+        )
+
+        // When
+        val wireframes = testedNumberPickerMapper.map(mockNumberPicker, fakeSystemInformation)
+
+        // Then
+        assertThat(wireframes).isEqualTo(
+            listOf(
+                expectedPrevLabelWireframe,
+                expectedTopDividerWireframe,
+                expectedSelectedLabelWireframe,
+                expectedBottomDividerWireframe,
+                expectedNextLabelWireframe
+            )
+        )
+    }
+
+    @Test
+    fun `M map the NumberPicker to a list of wireframes W map(){ value equals max }`() {
+        // Given
+        fakeValue = fakeMaxValue
+        whenever(mockNumberPicker.value).thenReturn(fakeValue)
+        val expectedSelectedLabelValue = "xxx"
+        val expectedPrevLabelValue = "xxx"
+        val expectedNextLabelValue = "xxx"
+        val expectedPrevLabelWireframe = MobileSegment.Wireframe.TextWireframe(
+            id = fakePrevLabelId,
+            x = fakeExpectedPrevLabelXPos,
+            y = fakeExpectedPrevLabelYPos,
+            width = fakeExpectedLabelWidth,
+            height = fakeExpectedLabelHeight,
+            textStyle = MobileSegment.TextStyle(
+                family = NumberPickerMapper.FONT_FAMILY,
+                size = fakeExpectedTextSize,
+                color = fakeExpectedNextPrevLabelHtmlColor
+            ),
+            textPosition = MobileSegment.TextPosition(
+                alignment = MobileSegment.Alignment(
+                    horizontal = MobileSegment.Horizontal.CENTER,
+                    vertical = MobileSegment.Vertical.CENTER
+                )
+            ),
+            text = expectedPrevLabelValue
+        )
+        val expectedTopDividerWireframe = MobileSegment.Wireframe.ShapeWireframe(
+            id = fakeTopDividerId,
+            x = fakeExpectedTopDividerXPos,
+            y = fakeExpectedTopDividerYPos,
+            width = fakeExpectedDividerWidth,
+            height = fakeExpectedDividerHeight,
+            shapeStyle = MobileSegment.ShapeStyle(
+                backgroundColor = fakeExpectedSelectedLabelHtmlColor
+            )
+        )
```

Comment:
since in all the tests here and in other file most of the properties set are the same, won't it be easier and shorter to introduce the methods which create these objects reading `fakeXXX` values, and then in the test, since it is still a `data` class, just use `.copy` call with setting properties specific for the test. Otherwise it is hard to follow, test setup is very long.